lib/labels.js

```diff
@@ -0,0 +1,254 @@
+          const code = res.statusCode;
+          if (
+            ((method === 'GET' || method === 'PATCH') && code === 200) ||
+            (method === 'POST' && code === 201) ||
+            (method === 'DELETE' && code === 204)
+          ) {
+            resolve({ data, res });
+          } else {
+            const err =
+              `${method} request to '${path}' ` +
+              `returned '${res.statusMessage}'`;
+            reject(err);
+          }
+        });
+      });
+      if (data) req.write(data);
+      req.once('error', reject);
+      req.end();
+    });
+  }
+
+  [getPath](repo, labelName = '') {
+    if (labelName) labelName = `/${encodeURIComponent(labelName)}`;
+    return `/repos/${repo}/labels${labelName}`;
+  }
+
+  // Get single label or all labels if `labelName` is not specified
+  // from `repo`
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   labelName <string>
+  async getFrom(repo, labelName) {
+    let path = this[getPath](repo, labelName);
+    if (labelName) {
+      const { data } = await this[request](path, 'GET');
+      return JSON.parse(data);
+    }
+    const result = [];
+    path += '?per_page=100';
+    while (path) {
+      const { data, res } = await this[request](path, 'GET');
+      result.push(...JSON.parse(data));
+
+      if (!res.headers.link) break;
+      path = parseLinkHeader(res.headers.link).get('next');
+    }
+    return result;
+  }
+
+  // Get single label or all labels if `labelName` is not specified
+  // from current repository
+  //   labelName <string>
+  async get(labelName) {
+    return this.getFrom(this.options.repo, labelName);
+  }
+
+  // Create label or labels in `repoOwner/repoName`
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   label <Object> | <Object[]>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>, optional
+  async createIn(repo, label) {
+    const path = this[getPath](repo);
+    if (Array.isArray(label)) {
+      return Promise.all(
+        label.map(label => this[request](path, 'POST', JSON.stringify(label)))
+      ).map(({ data }) => JSON.parse(data));
+    } else {
+      const { data } = await this[request](path, 'POST', JSON.stringify(label));
+      return JSON.parse(data);
+    }
+  }
+
+  // Create label or labels in current repository
+  //   label <Object> | <Object[]>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>, optional
+  async create(label) {
+    return this.createIn(this.options.repo, label);
+  }
+
+  // Get used labels from `repo`
+  //   repo <string> e.g. `repoOwner/repoName`
+  async getUsedLabels(repo) {
+    const { data } = await this[request](
+      `/repos/${repo}/issues?state=all`,
+      'GET'
+    );
+    const result = new Map();
+    JSON.parse(data).forEach(issue =>
+      issue.labels.forEach(label => {
+        const name = label.url.split('/').pop();
```

Comment:
Why not just use `label.name`?
```suggestion
        const { name } = label;
```

Reply:
It will already be encoded and this also solves the issue mentioned in https://github.com/metarhia/tools/pull/5#discussion_r296176442.

Reply:
In the comment you mentioned, I explicitly said that it wouldn't work:

> The fact that you are using part of the URL returned by the API above doesn't help since the way the search works is different.